magefile.go

```diff
@@ -1295,3 +1321,123 @@ func shouldBuildAgent() bool {
 	}
 	return ret
 }
+
+// Pre-requisite: user must have the gcloud CLI installed
+func authGCP(ctx context.Context) error {
+	// Check if gcloud CLI is installed
+	const cliName = "gcloud"
+	cmd := exec.CommandContext(ctx, "which", cliName)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("%s CLI is not installed: %w", cliName, err)
+	}
+
+	// Check if user is already authenticated
+	for authSuccess := false; !authSuccess; {
+		cmd = exec.CommandContext(ctx, cliName, "auth", "list", "--filter=status:ACTIVE", "--format=json")
```

Comment:
Should we check (if possible) to determine that they are authenticated with an elastic account that they are using the correct platform ingest project?

Reply:
Yes, good idea.  I think we can at least check the domain of the email address they're authenticated with, which should be good enough to say if they're authenticated with an elastic account.  I'm not sure if there's a better / more structured way than that.

I think we can check the GCP project as well.  Do you recall which project we're supposed to be using?  If I sign in, I see `elastic-obs-integrations-dev` as one of the options for me.  I think that's it but maybe there's another one for Ingest?  What do you see when you sign in?

Reply:
Thinking about this some more, do we want to add these extra checks, considering this target is OSS?  Say an external (non-Elastic) contributor wanted to write an integration test and wanted to run their test locally, these restrictions would get in their way.  Also, as far as the test runner is concerned, does it need to really care about whether the authenticated account is an `@elastic.co` email address or if it's using the `elastic-obs-integrations-dev` GCP project?

Reply:
I implemented the extra checks in 029b8d59ca5476e35d875a673def0e62e282ee00 but I'm really not sure we should add these, given that this is an OSS project repo (see previous comment for more detail).  WDYT?

Reply:
I am torn. I think it will be helpful for us, and we will be the main users. I also don't know how many outside of Elastic will want to run integration tests on there own GCP account, feel like they will more likely rely on CI. But being an OSS project I could see that we might not want it. I am on the fence.

Reply:
How about we expose a boolean environment variable, something like `TEST_INTEG_AUTH_GCP_SKIP_ELASTIC_CHECK`.  Unless this variable is set to a `true`-y value (basically anything [`strconv.ParseBool`](https://pkg.go.dev/strconv#ParseBool) would evaluate to `true`), we'd perform the additional Elastic-specific checks. 

In other words, by default, the Elastic-specific checks will be performed but if a non-Elastic contributor wants to run this target locally, they have a way to skip past these Elastic-specific checks.

WDYT?

Reply:
I implemented the skip env var idea (see previous comment) in 10eeeea9d.  Let me know what you think.

Reply:
Yeah. I like the env variable.